lua/lspconfig/util.lua

```diff
@@ -294,7 +243,52 @@ function M.strip_archive_subpath(path)
   return path
 end
 
---- Functions that can be removed once minimum required neovim version is high enough
+--- Public functions that can be deprecated once minimum required neovim version is high enough
+
+local function is_fs_root(path)
+  if iswin then
+    return path:match '^%a:$'
+  else
+    return path == '/'
+  end
+end
+
+-- Traverse the path calling cb along the way.
+local function traverse_parents(path, cb)
```

Comment:
how about this 🤔

```lua
local function traverse_parents(path, callback)
    local real_path = vim.loop.fs_realpath(path)
    if not real_path then
        return nil, nil
    end

    if type(callback) ~= "function" then
        error("callback must be a function")
    end

    local dir = real_path

    while dir do
        local parent = vim.fs.dirname(dir)
        if not parent or parent == dir then
            break
        end

        local should_stop = callback(parent, real_path)
        if should_stop then
            return parent, real_path
        end

        if is_fs_root(parent) then
            break
        end

        dir = parent
    end

    return nil, nil
end
```

Reply:
Do we really need to refactor this function? We'll remove it once `is_descendant` is removed anyway, so think we can treat this function as frozen.

Reply:
No need as far as I remember this function is not used in lspconfig 😄